src/IoInteractions.hs

```diff
@@ -0,0 +1,29 @@
+module IoInteractions where
+
+import Data.Word
+import Data.Time.Clock
+import Data.Time.Calendar
+import LoadEnv
+import System.Environment
+import Database
+
+loadConnInfoFromEnv :: IO ConnInfo
+loadConnInfoFromEnv = do
+  loadEnv
+  ConnInfo
+    <$> fromEnv "HOST"
+    <*> (read @Word16 <$> fromEnv "PORT")
+    <*> fromEnv "USER"
+    <*> fromEnv "PASS"
+    <*> fromEnv "DB"
+
+fromEnv :: String -> IO String
+fromEnv tag = do
+  v <- lookupEnv tag
+  maybe (error $ msg tag) return v
+  where msg t = "Could not load var: " ++ t
+
+getSystemDay :: IO Day
+getSystemDay = do
```

Comment:
What about this: `getSystemDay = utctDay <$> getCurrentTime`?

Reply:
Looks great, did not realice it before, thanks.